tests/viz_tests.py

```diff
@@ -634,3 +634,109 @@ def test_rose_vis_get_data(self):
             ],
         }
         self.assertEqual(expected, res)
+
+
+class BaseDeckGLVizTestCase(unittest.TestCase):
+
+    def test_get_metrics(self):
+        form_data = {'size': 'large'}
+        datasource = {'type': 'table'}
```

Comment:
I think `datasource` expects a derivative of `connectors.base.models.BaseDatasource`, I'm surprised this works.